expected-failures/pandoc.txt

```diff
@@ -1,3 +1,6 @@
+Found .cabal file /build/pandoc-2.14.0.3/pandoc.cabal, but it did not mention Setup.hs
+Found .cabal file /build/pandoc-2.14.0.3/pandoc.cabal, but it did not mention prelude/Prelude.hs
+Found .cabal file /build/pandoc-2.14.0.3/pandoc.cabal, but it did not mention src/Text/Pandoc/Data.hs
```

Comment:
This file is not detected as it is behind a feature flag: [see here](https://github.com/jgm/pandoc/blob/227f675cc46a680d4b8f2fd71e0c4cbb92a3fd0b/pandoc.cabal#L512)

We could use [`ignoreConditions`](https://hackage.haskell.org/package/Cabal-3.6.1.0/docs/Distribution-Types-CondTree.html#v:ignoreConditions) to also detect modules like this.

Reply:
I could think of a disadvantage when following this approach, so I went with it in 20532c816d2e90384a08060ba6c85fa081bdcc8c.

Reply:
Yes, but I imagine this to be very rare, especially for parsing-related extensions which are relevant for Ormolu. But I could write a variant of `ignoreConditions` which only merges `exposed-modules` etc.

---

Somewhat related: Even right now, a module could be part of multiple components with differing `default-extensions` (in #759, we decided to pick an arbitrary component in this case).

Reply:
What I'm thinking about is that if default extensions can be conditional and also modules can be conditional then it makes sense to check if there are groups of extensions/modules that share the same condition. This can get pretty hairy quickly though, as I imagine "or" and "and" can be used in conditions...

Reply:
I think what you have right now is good enough. Let's approach this iteratively and address when/if real-world cases are discovered and reported.